### ISSUE TITLE
[gemm_v2] MIOpen API needs to change when int8x4 was removed in rocBLAS

### ISSUE THREAD

**Issue by junliume** [ROCmSoftwarePlatform/MIOpen#2226]
Deprecation for rocblas_gemm_flags_pack_int8x4 was announced in the rocBLAS major version rocBLAS 3.0  here: https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/develop/docs/API_Reference_Guide.rst#removal-of-rocblas_query_int8_layout_flag 

So for MIOpen this line needs to be changed for a rocBLAS version upper limit:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/9798115d51eb2fbf799026e0f0071de509826b6b/src/gemm_v2.cpp#L65

We are asking rocBLAS to provide that upper limit

**Comment by junliume** [comment]
Feedback from rocBLAS team:
```
rocBLAS version 4 or greater removes rocblas_gemm_flags_pack_int8x4. 
rocBLAS version 3 or earlier has rocblas_gemm_flags_pack_int8x4. 
You only need to check the rocBLAS major version number. 
```

**Comment by atamazov** [comment]
@junliume I think https://github.com/ROCmSoftwarePlatform/MIOpen/labels/IMPACTS_API should be removed.

**Comment by atamazov** [comment]
@junliume @JehandadKhan This needs to be reopened with https://github.com/ROCmSoftwarePlatform/MIOpen/labels/bug.

The simple removal of deprecated flags leads to correctness issues in GemmFwd solvers with INT8x4 datatypes. The simplest examples of failing configs:

```bash
./bin/test_conv2d --int8 --output_type int32 --vector_length 4 --int8_vectorize 1
./bin/test_conv2d --int8 --output_type float --vector_length 4 --int8_vectorize 1
```

I am working on this.

**Comment by atamazov** [comment]
@junliume The decision is to remove int8x4 support, so failing tests is no longer a bug, but something of a quality issue. ~The real issue will show up as build errors (with ROCm 6.0 RC).~

**Comment by atamazov** [comment]
@junliume Resolved via #2441, let's close this.
